cloud/clouds.go

```diff
@@ -58,6 +58,10 @@ const (
 	// CertificateAuthType is an authentication type using certificates.
 	CertificateAuthType AuthType = "certificate"
 
+	// HTTPSigAuthType is an authentication type that uses HTTP signatures:
+	// https://tools.ietf.org/html/draft-cavage-http-signatures-06
+	HTTPSigAuthType AuthType = "httpsig"
```

Comment:
interesting to read through the spec a bit.
That seems like a really bad algorithm, tbh.
The problem is that the spec doesn't mandate things like "include a signing the content of the body or a digest of the body" in the headers that get signed.
So while it is possible to create a safe implementation that *does* sign enough to be good. The standard also lets you write perfectly valid clients and servers that say "only sign the date". In which case anyone who gets a copy of that date, can forge any number of requests on behalf of that client, as long as they pretend the request was sent at a given date. (presumably valid server implementations might reject requests that are 'old enough' ?)

I'm not a big fan of security specs where you can easily write a conforming implementation that is completely insecure. :)